CRM/Sqltasks/Task.php

```diff
@@ -696,4 +707,13 @@ public static function getAllFiles() {
   public static function getLastFile() {
     return end(self::$files);
   }
+
+  /**
+   * Replace all tokens in the string with data from the value
+   */
+  protected function resolveTokens(&$string, $value, $token_value) {
+    $token = preg_quote('{' . "$token_value") . '(\|([^\}]+?))?' . preg_quote('}');
+
+    $string = preg_replace("/([^\{])?$token/", "\${1}$value", $string);
```

Comment:
This seems more complex than the simple `str_replace` in `CRM_Sqltasks_Action::resolveTokens`, is this needed for some reason I'm probably missing, or could we swap to the simpler one?

Reply:
I could swap to the simpler one.

Reply:
Sounds good, let's go with that.